Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -94,22 +99,156 @@ function buildFilePath(localPathRoot: string, fileEnding: string, fileNameAndPat
     return extractSource;
 }
 
-function getFileEnding(file: string): string {
-    let fileEnding = '';
-
-    if (file.endsWith('.tar')) {
-        fileEnding = '.tar';
-    } else if (file.endsWith('.tar.gz')) {
-        fileEnding = '.tar.gz';
-    } else if (file.endsWith('.zip')) {
-        fileEnding = '.zip';
-    } else if (file.endsWith('.7z')) {
-        fileEnding = '.7z';
+/**
+ * Return file ending if it is supported. Otherwise throw an error.
+ * @param JDKfile Path to a file.
+ */
+function getSupportedFileEnding(file: string): string {
+    for (const fileEnding of supportedFileEndings) {
```

Comment:
I guess using the [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) method makes the code more readable.

```
const fileExtension: string = supportedFileEndings.find((fileEnding) => file.endsWith(fileEnding)); 

if (fileExtension) {
    return fileExtension;
} else {
    throw new Error(taskLib.loc('UnsupportedFileExtension'));
}
```

Reply:
Thanks for the suggestion! Changed it.